Fixed manage students period select dropdown updating issues

hirokiterashima opened the pull request:
## Changes
- Period select correctly updates number of teams
- Can view team that is moved to an empty period
- Moved empty teams to bottom of manage periods view
- Cleaned up code

## Test
- Move a team to another period. The period select should update the number of teams in each period.
- Move a team to another period that has no teams. You should be able to go to that period and see the team.

Closes #271

hirokiterashima committed:
Fixed manage students period select dropdown updating issues.
- period select correctly updates number of teams
- can view team that is moved to an empty period
- moved empty teams to bottom of manage periods view
- cleaned up code

#271

geoffreykwan reviewed (changes requested):
I found a couple bugs.

- An empty team shows up in all the periods.

- Can't move student into an empty team.

- For an empty team you can click on the "Change Period" link but the period select in the popup does not work. Maybe disable the "Change Period" link for empty teams since it doesn't really make sense?

- Can't go to period with 0 teams. Should be able to go to Period with 0 teams to see empty teams.

hirokiterashima committed:
Show period in select-drop down even with no teams.
Fix: drag students to empty teams.
Fix: show empty teams in correct period.
Hide change period link for empty teams.
#271

hirokiterashima commented:
@geoffreykwan good catch. I addressed all those issues.

geoffreykwan reviewed (approved):
Looks good.
